goas/zone/risc0_proofs/spend_zone_funds/src/main.rs

```diff
@@ -0,0 +1,111 @@
+/// Zone Funds Spend Proof
+///
+/// Our goal: prove the zone authorized spending of funds
+use cl::merkle;
+use cl::nullifier::{Nullifier, NullifierNonce, NullifierSecret};
+use cl::partial_tx::PtxRoot;
+use proof_statements::zone_funds::{SpendFundsPrivate, SpendFundsPublic};
+use risc0_zkvm::guest::env;
+use sha2::{Digest, Sha256};
+
+fn main() {
+    let SpendFundsPrivate {
+        in_zone_funds,
+        out_zone_funds,
+        zone_note,
+        spent_note,
+        spend_event,
+        in_zone_funds_ptx_path,
+        in_zone_funds_cm_path,
+        out_zone_funds_ptx_path,
+        zone_note_ptx_path,
+        spent_note_ptx_path,
+        spend_event_state_path,
+        in_zone_funds_nf_sk: _,
+    } = env::read();
+
+    let in_zone_funds_cm = in_zone_funds.commit();
+    let in_zone_funds_leaf = merkle::leaf(&in_zone_funds_cm.to_bytes());
+    let cm_root = merkle::path_root(in_zone_funds_leaf, &in_zone_funds_cm_path);
```

Comment:
I think this is the wrong commitment here, you are using the input commitment in this path check instead of the UTXO commitment.



```suggestion
    let cm_root = merkle::path_root(in_zone_funds.to_output_witness().commit_note(), &in_zone_funds_cm_path);
```

Reply:
right, although I find `InputWitness::commit() -> Input` a bit misleading as name, should it be something like `to public`?

Reply:
I think it's useful to have a common interface of `secret witness --[commit()]---> public commitment` across all the different CL objects.

I'm not sure how to make this more clear. The good new is that this would never validate so you will quickly find the bug.